examples/performance/arithmetic/arithmetic.cxx

```diff
@@ -14,57 +14,61 @@ typedef std::chrono::duration<double> Duration;
 using namespace std::chrono;
 
 class Arithmetic : public PhysicsModel {
-protected: 
+protected:
   int init(bool restarting) {
-    
+
     Field3D a = 1.0;
     Field3D b = 2.0;
     Field3D c = 3.0;
 
     Field3D result1, result2, result3, result4;
 
     // Using Field methods (classic operator overloading)
-    
-    result1 = 2.*a + b * c;
 
-    SteadyClock start1 = steady_clock::now();
     result1 = 2.*a + b * c;
-    Duration elapsed1 = steady_clock::now() - start1;
-    
-    // Using C loops
-    result2.allocate();
-    BoutReal *rd = &result2(0,0,0);
-    BoutReal *ad = &a(0,0,0);
-    BoutReal *bd = &b(0,0,0);
-    BoutReal *cd = &c(0,0,0);
-    SteadyClock start2 = steady_clock::now();
-    for(int i=0, iend=(mesh->LocalNx*mesh->LocalNy*mesh->LocalNz)-1; i != iend; i++) {
-      *rd = 2.*(*ad) + (*bd)*(*cd);
-      rd++;
-      ad++;
-      bd++;
-      cd++;
+
+    Duration elapsed1,elapsed2,elapsed3,elapsed4;
+    for (int ik=0;ik<1e2;++ik){
+      SteadyClock start1 = steady_clock::now();
+      result1 = 2.*a + b * c;
+      elapsed1 = steady_clock::now() - start1;
```

Comment:
Doesn't this end up just recording the time for the last iteration of the outer loop (or is that what's intended, with 99 trial runs)?

Reply:
yes, this is intentionally.
100 is maybe an overkill ... but it doesn't hurt.
Maybe switch to `min`?
`+=` would include the cache issues from the old version.

Reply:
Ah OK I see, so it is intended for warmup rather than averaging. A comment may help here. 

If we wanted to also add averaging in how would you recommend we do it in this case? A separate loop after the warmup?